ocaml/testsuite/tests/typing-modes/legacy.ml

```diff
@@ -26,7 +26,7 @@ let (foo @ portable) () =
 Line 2, characters 20-21:
 2 |     let _ = (module M : S) in
                         ^
-Error: The module M is unportable, so cannot be used inside a closure that is portable.
+Error: Modules are unportable, so cannot be used inside a closure that is portable.
```

Comment:
This will be fixed after rebasing (I didn't rebase for incremental reviewing)

Reply:
Ugh I thought this was fixed in a previous PR, so I didn't change in this PR because I thought we just need a rebase. But turned out `env.ml` has its own copy of mode printing. I will fix this; in the long run we should have a single copy of mode-printing.